model/src/lectures/lectureDoc.ts

```diff
@@ -31,16 +31,27 @@ namespace LectureDisplay {
         quizModel: Quizzes.Quiz;
     }
 
+    export interface Question {
+        text: string,
+        //userID: string,
+        anonym: boolean,
+        likes: number,
+        postedTime: Timestamp,
+    }
+
     export interface Lecture {
         pdfUri: string;
         nbOfPages: number;
         availableToStudents: boolean;
         audioTranscript: { [pageNumber: number]: string; };
+        questions: Question[];
```

Comment:
I am not sure that keeping the questions as an array inside the Lecture document is the best option, as it can create concurrency and scalability problems down the line, it is also useful to be able to refer to one specific question (index is a weak pointer). Maybe we can change it to be a sub-collection of the Lecture ? Let me know what you think.

Reply:
Ok I understand, however, I am just having some trouble trying to comprehend how to manage my model in order for it to be like that. How do I handle the connection between the question and the Lecture? I assume that if it becomes a sub-collection of the Lecture, there is no longer a need to keep the variable questions inside Lecture. Then is it just that they are connected only inside Firebase? For example having createQuestion create a new question inside a sub-collection of the currentLecture in Firebase?

Reply:
Hello,
Yes exactly, `Lecture` loses its `questions` field.
The cloud function `createQuestion` should then ask for the document `id` of the lecture to be able to add it to the sub-collection.